File: 02-moderate/first_non-repeated_character/first.non-repeated.character.py
# Reuben Thorpe (2016), CodeEval [First Non-Repeated Character v1.0]
from sys import argv


with open(argv[1], "r") as in_file:
    for line in in_file:
        for char in line:
            if line.count(char) == 1:
                print(char)
                break

File: code.eval/02-moderate/first_non-repeated_character/first.non-repeated.character.cpp
// Reuben Thorpe (2016), CodeEval [First Non-Repeated Character v1.1]
#include <iostream>
#include <fstream>
#include <algorithm>

using namespace std;


int main(int argc, char* argv[]) {
  ifstream in_file(argv[1]);
  string line;

  while(getline(in_file, line)) {
    for (auto c: line) {
      if (count(line.begin(), line.end(), c) == 1) {
        cout << c << '\n';
        break;
      }
    }
  }

  return(0);
}
